### ISSUE TITLE
Custom markers are hard to customize

### ISSUE THREAD

**Issue by karlmtc** [oxyplot/oxyplot#487]
I've been trying to use a line serie and a scatter serie to represent the sun position over the hours of the day. The main issue that I have is that I'd like to customize the scatter point's marker (just one, that will be the sun circle) but I haven't found an easy way to achieve this.

What I'd like to do from a WPF perspective is to set the MarkerType to custom, and then set the let's call it "MarkerTemplate" (which i think should be a DataTemplate or some sort of container), instead of the marker outline, that I'm not able to understand how it works. This way I could choose to set the marker as an image of the sun, a path that I can get from an SVG or any other control.

Any thoughts on this?

**Comment by objorke** [comment]
I understand this as a new feature
- show markers defined by image
- show markers defined by an svg
- show markers defined by delegate (e.g. Action<IRenderContext>)?

**Comment by objorke** [comment]
I don't think we should add svg parsing into this library (keep it simple...)

**Comment by karlmtc** [comment]
Sorry if I didn't explain correctly, but I wasn't asking to add svg parsing or support. What I meant is to add support for a DataTemplate to the marker. That way I could do something like:


<oxy:ScatterSeries MarkerType="Custom">
    <oxy:ScatterSeries.Resources>
        <StreamGeometry x:Key="SunPath">47.853" Height="47.853" Canvas.Left="14.0735" Canvas.Top="14.0735" Stretch="Fill" Fill="#FF000000" Data="F1 M 61.9265,41.022L 53.0007,43.0713C 53.6708,41.0949 53.9557,38.9728 53.7924,36.8188L 61.9265,41.022 Z M 52.7817,57.0555L 45.0225,52.1939C 46.9628,51.2366 48.6434,49.9111 50.0017,48.3301L 52.7817,57.0555 Z M 34.978,61.9265L 32.9287,53.0007C 34.9051,53.6708 37.0272,53.9557 39.1812,53.7924L 34.978,61.9265 Z M 18.9445,52.7817L 23.8061,45.0225C 24.7634,46.9629 26.0889,48.6435 27.6699,50.0017L 18.9445,52.7817 Z M 14.0735,34.978L 22.9993,32.9288C 22.3292,34.9052 22.0443,37.0272 22.2076,39.1812L 14.0735,34.978 Z M 23.2183,18.9445L 30.9775,23.8062C 29.0371,24.7634 27.3565,26.0889 25.9983,27.6699L 23.2183,18.9445 Z M 41.022,14.0735L 43.0712,22.9993C 41.0948,22.3292 38.9728,22.0443 36.8188,22.2076L 41.022,14.0735 Z M 57.0555,23.2183L 52.1938,30.9775C 51.2366,29.0371 49.9111,27.3566 48.3301,25.9983L 57.0555,23.2183 Z M 38,25.3333C 44.9956,25.3333 50.6667,31.0044 50.6667,38C 50.6667,44.9956 44.9956,50.6667 38,50.6667C 31.0044,50.6667 25.3333,44.9956 25.3333,38C 25.3333,31.0044 31.0044,25.3333 38,25.3333 Z </StreamGeometry>
    </oxy:ScatterSeries.Resources>

    <oxy:ScatterSeries.MarkerTemplate>
        <DataTemplate>
            <Path Width="20" Height="20" Data="{StaticResource SunPath} Fill="Black" />
        </DataTemplate>
    </oxy:ScatterSeries.MarkerTemplate>

</oxy:ScatterSeries>


What I'm trying to explain with this, is that the positioning should remain as it was, but the visualization should be templated with the given template. I don't know how portable is this approach, but it's the ideal for WPF.

**Comment by objorke** [comment]
This would only be possible on the XAML based platforms, I think (WPF, SL, Windows apps, Xamarin Forms..) - and I don't see an easy way to implement it since the rendering is all handled in portable code...

**Comment by objorke** [comment]
this could also be part of the feature:
- show custom markers defined by a path

**Comment by karlmtc** [comment]
A "quick" approach would be to provide a geometry and convert it to DataPoints (the MarkerOutline) but I don't think it would be the best approach since it's kind of limited to only drawing paths. 
I'd rather think a solution to make a portable approach to draw objects and specify in the concrete platform how to draw them. 

I mean abstract that drawing process in the portable library and let the concrete platform assembly define how. Don't know if that makes sense...

In the portable library do something like:

public abstract void DrawMarker();

In the concrete platform:

public override void DrawMarker()
{
    //do concrete stuff with datatemplates
}

Does this make any sense? I didn't have much time to look at the current code. Nice job so far, by the way.

**Comment by objorke** [comment]
The current design is to keep the implementation on each platform very simple and do all the drawing in portable code. I think it should be easy to support paths, images and delegated drawing from portable code. Introducing support for XAML data templates will make the solution much more complex, I guess.
